### PR TITLE
Supported snapshot for the remote iscsi and ceph images

### PR DESCRIPTION
**Pull request by zhencliu**
1. Added cfg/remote_image_snap.cfg
  2. TC change for remote images

Signed-off-by: Zhenchao Liu <zhencliu@redhat.com>

ID: 1778550 1778552

### CODE REVIEW

**Comment by hellohellenmao** [comment]
@zhencliu `git am` failed as below, check, please.

# git am 2017.patch 
Applying: Supported snapshot for the remote iscsi and ceph images
/home/kar/workspace/var/lib/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/.git/rebase-apply/patch:53: trailing whitespace.
            enable_iscsi_snA = no 
warning: 1 line adds whitespace errors.

**Comment by zhencliu** [comment]
@lolyu  Can U please help to review the tc change, thanks much.

File: qemu/tests/qemu_disk_img_info.py [hunk 1]
```diff
@@ -50,9 +49,38 @@ def run(test, params, env):
...
-    params.update(
-        {"image_name_%s" % base_image: params["image_name"],
-         "image_format_%s" % base_image: params["image_format"]})
+
+    update_params = {
+        "image_name_%s" % base_image: params["image_name"],
+        "image_format_%s" % base_image: params["image_format"]
+    }
+
+    optval = (lambda opt, img, p, default: p.get('%s_%s' % (opt, img),
```
Comment (lolyu) [inline qemu/tests/qemu_disk_img_info.py]: Please define a function instead of assigning `lambda` function to a variable.

Comment (zhencliu) [reply qemu/tests/qemu_disk_img_info.py]: Confirmed with lolyu, for the pep8 checker ignored the warning when use (), leave code unchanged.

**Comment by lolyu** [comment]
test result, please.

**Comment by zhencliu** [comment]
@hellohellenmao 
--testcase=remote_image_snapshot

**Comment by zhencliu** [comment]
> test result, please.

Tingting will paste

**Comment by hellohellenmao** [comment]
Passed over ISCSI:

`# python ConfigTest.py --testcase=remote_image_snapshot --machine=q35 --guestname=RHEL.8.1.1 --imageformat=luks --storage-backend-conf=/tmp/iscsi.cfg --clone=no --customsparams='qemu_force_use_drive_expression=no'`

`(1/2) Host_RHEL.m8.u2.product_av.iscsi_direct.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.1.x86_64.io-github-autotest-qemu.remote_image_snapshot.to_local.q35: PASS (202.55 s)
 (2/2) Host_RHEL.m8.u2.product_av.iscsi_direct.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.1.x86_64.io-github-autotest-qemu.remote_image_snapshot.to_remote.q35: PASS (199.20 s)`

**Comment by hellohellenmao** [comment]
Passed over ceph.

`# python ConfigTest.py --testcase=remote_image_snapshot --machine=q35 --guestname=RHEL.8.1.0 --imageformat=raw --storage-backend-conf=/tmp/ceph.cfg --clone=no --customsparams='qemu_force_use_drive_expression=no'INFO | Setting up symlink target 'images`

` (1/2) Host_RHEL.m8.u2.product_av.ceph.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.remote_image_snapshot.to_local.q35: PASS (246.25 s)
 (2/2) Host_RHEL.m8.u2.product_av.ceph.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.remote_image_snapshot.to_remote.q35: PASS (734.90 s)
`

**Comment by CongLi** [comment]
@luckyh Could you please help review this PR ?

Thanks.

**Comment by zhencliu** [comment]
> @luckyh Could you please help review this PR ?
> 
> Thanks.

it depends on the vt patch https://github.com/avocado-framework/avocado-vt/pull/2361

**Pull request merged**